state-chain/chains/src/sol/api.rs

```diff
@@ -401,6 +399,20 @@ impl<Environment: SolanaEnvironment> SolanaApi<Environment> {
 		let compute_price = Environment::compute_price()?;
 		let durable_nonce = Environment::nonce_account()?;
 
+		// Get the Address lookup tables. Chainflip's ALT is proceeded with the User's.
+		let external_alts = ccm_additional_data
+			.alt_addresses()
+			.map(|alt_addresses| Environment::get_address_lookup_tables(alt_addresses))
+			.transpose()
+			.inspect_err(|_| {
+				Environment::recover_durable_nonce(durable_nonce.0);
+			})?
```

Comment:
Note here, we have to remember to revert any state if we exit with an error. 

So I moved the ALT lookup (which *takes* the data) after all the `?` points and added a call to recover the nonce if it errors. 

We should not have to rely on remembering this... I think ideally all these tx construction methods should be transactional - then we won't need to remember to eg. `recover_durable_nonce`. I'll raise an issue.

Reply:
(https://linear.app/chainflip/issue/PRO-2238/make-transaction-building-transactional)